src/Microsoft.AspNet.Mvc.ModelBinding/Validation/DataAnnotationsClientModelValidatorOfTAttribute.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
+{
+    /// <summary>
+    /// An implementation of <see cref="IClientModelValidator"/> which understands data annotation attributes.
+    /// </summary>
+    /// <typeparam name="TAttribute">The type of the attribute.</typeparam>
+    public abstract class DataAnnotationsClientModelValidator<TAttribute> : IClientModelValidator
+        where TAttribute : ValidationAttribute
+    {
+        /// <summary>
+        /// Create a new instance of <see cref="DataAnnotationsClientModelValidator{TAttribute}"/>.
+        /// </summary>
+        /// <param name="attribute">The <typeparamref name="TAttribute"/> instance to validate.</param>
+        public DataAnnotationsClientModelValidator(TAttribute attribute)
+        {
+            Attribute = attribute;
+        }
+
+        /// <summary>
+        /// Gets the type of the <typeparamref name="TAttribute"/> associated with this instance.
```

Comment:
no, property is not of type `Type`.  `Gets the <typeparamref name="TAttribute"/> instance to validate.` or maybe just `Gets the <typeparamref name="TAttribute"/> instance.`